web/shapeworks/src/views/Data.vue

```diff
@@ -30,21 +39,96 @@ export default defineComponent({
 
         const mini = ref(false);
         const search = ref('');
-        const headers = [
-            {text: 'ID', sortable: true, value: 'id'},
-            {text: 'Type', sortable: true, value: 'type'},
-            {text: 'File Name', sortable: true, value: 'file'},
-        ]
+        const headers = ref()
+        const rows = ref<number>(1);
+        const cols = ref<number>(1);
+        const renderData = ref<ShapeData[]>([]);
+
+
+        async function fetchDataObjects() {
+            if (!selectedSubject.value) return;
+            loadingState.value = true;
+            allDataObjectsForSubject.value = await getDataObjectsForSubject(selectedSubject.value.id)
+            renderData.value = []
+            loadingState.value = false;
+        }
 
         function shortFileName(file: string) {
             const split = file.split('?')[0].split('/')
             return split[split.length-1]
         }
 
+        function displayLocation(id: number){
+            const selectedObjectIds = selectedDataObjects.value.map((obj) => obj.id)
+            if(selectedObjectIds.includes(id)){
+                const index = selectedObjectIds.indexOf(id);
+                const i = index % cols.value;
+                const j = Math.floor(index / cols.value)
+
+                return "ABCDE".charAt(i)+(j+1).toString();
+            }
+            return '';
+        }
+
+        function updateDisplayLocations() {
+            allDataObjectsForSubject.value = allDataObjectsForSubject.value.map(
+                (obj) => Object.assign(obj, {'display': displayLocation(obj.id)})
+            )
+        }
+
+        function setTableHeaders() {
+            headers.value = [
+                {text: 'ID', sortable: true, value: 'id'},
+                {text: 'Type', sortable: true, value: 'type'},
+                {text: 'File Name', sortable: true, value: 'file'},
+
+            ]
+            if(selectedDataObjects.value.length > 0){
+                headers.value.push(
+                    {text: 'Display', sortable: true, value: 'display'}
+                )
+            }
+        }
```

Comment:
This looks like it would be simpler if `headers` were a computed property, then you wouldn't need to worry about calling `setTableHeaders` at all.

Reply:
Changed with [94f46ba](https://github.com/girder/shapeworks-cloud/pull/134/commits/94f46ba50a397c8d47d605b21528ed46c71fd0ae)